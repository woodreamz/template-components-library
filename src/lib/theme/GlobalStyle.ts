import { createGlobalStyle } from 'styled-components';
import reset from 'styled-reset';

/**
 * Application Global Style.
 */
const GlobalStyle = createGlobalStyle`
  ${reset};
  html,
  body,
  #root {
    height: 100%;
    width: 100%;
    font-weight: ${({ theme }) => theme.fonts.regular};
    box-sizing: border-box;
    background-color: ${({ theme }) => theme.primaryBackground};
    font-family: ${({ theme }) => theme.text.fontFamily};
    font-size: ${({ theme }) => theme.fonts.defaultSize};
    color: ${({ theme }) => theme.text.primary};
  }

  * {
    font-family: ${({ theme }) => theme.text.fontFamily};
  }
`;

export default GlobalStyle;
