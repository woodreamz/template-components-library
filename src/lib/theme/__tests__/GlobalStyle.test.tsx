import { render } from '../../../tests';
import GlobalStyle from '../GlobalStyle';

describe('<GlobalStyle />', () => {
  it('should render correctly', () => {
    const { container } = render(<GlobalStyle />);
    expect(container).toMatchSnapshot();
  });
});
