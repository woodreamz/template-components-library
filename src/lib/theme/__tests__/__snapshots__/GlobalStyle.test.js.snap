// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GlobalStyle /> Should render correctly 1`] = `<div />`;
